saleor/graphql/checkout/tests/test_checkout_lines.py

```diff
@@ -512,6 +512,26 @@ def test_checkout_lines_update_variant_quantity_over_avability_stock(
     assert data["checkout"]["lines"][0]["quantity"] == variables["lines"][0]["quantity"]
 
 
+def test_checkout_lines_delete_with_by_zero_quantity_when_variant_out_of_stock(
+    user_api_client, checkout_with_item
+):
+    checkout = checkout_with_item
+    line = checkout.lines.first()
+    variant_id = graphene.Node.to_global_id("ProductVariant", line.variant.pk)
+    line.quantity = 0
```

Comment:
Shouldn't be set to something different than 0 here? Assuming that we have some quantity assigned in Checkout. Then we call mutation to set it to 0. 
Also, you should modify the assigned stock to this variant and modify it to 0.